cyences_app_for_splunk/appserver/static/privilege_user_role_editor.js

```diff
@@ -0,0 +1,289 @@
+require([
+    "splunkjs/mvc",
+    "underscore",
+    "jquery",
+    'splunkjs/mvc/tableview',
+    '../app/cyences_app_for_splunk/splunk_common_js_v_utilities'
+], function (
+    mvc,
+    _,
+    $,
+    TableView,
+    SplunkCommonUtilities
+) {
+
+    const USER_INVENTORY_TABLE_SEARCH_ID = 'base_summary';
+    const USER_INVENTORY_TABLE_ID = 'user_inventory_main_tbl';   // NOTE - CSS contains this hard-coded value.
+    const DEFAULT_USER_ROLE = '(unchanged)';
+    
+    let AVAILABLE_USER_ROLES = [DEFAULT_USER_ROLE, 'Administrator', 'admin', 'Dcadmin', 'Root'];
+    let selected_user_row_events = [];
+    let all_users = [];
+
+    // creates the div having diff options to select/edit the raws if the search contains the results.
+    let userEventSearchManager = new SplunkCommonUtilities.VSearchManagerUtility(
+        function(data){
+            if (data == null){
+                all_users = [];
+                $("#bulk_edit_container").remove();
+                return;
+            }
+            let uuid = data.fields.indexOf("uuid");
+            all_users = _.map(data.rows, function (e) { return e[uuid]; });
+            $("#bulk_edit_container").remove();
+            $(`#${USER_INVENTORY_TABLE_ID}`).parent().before($("<div />").attr('id', 'bulk_edit_container').addClass("bulk_edit_container").addClass('panel-element-row'));
+            var links = _.template('<a href="#" id="bulk_edit_select_all">Select All</a> | <a href="#" id="bulk_edit_selected">Edit Selected</a> | <a href="#" id="bulk_edit_all">Edit All <%-nr_users%> Matching Users</a> | <a href="#" id="bulk_edit_clear">Reset Selection</a>', { nr_users: all_users.length });
+            $("#bulk_edit_container").html(links);
+            $("#bulk_edit_container").show();
+        },
+        function(errorProperties){
+            console.log("User inventory table search failed.");
+        }
+    );
+    userEventSearchManager.searchById(USER_INVENTORY_TABLE_SEARCH_ID);
+
+    function restartUserEventSearch(){
+        userEventSearchManager.startSearch();
+    }
+
+    // create the search to execute when user click on save button of the popup.
+    function runUserInventoryUpdaterSearch(entries_to_update, callBackFunction){
+        // Splitting array into multiple queries in case bulk update contains many results, it may result into failure
+        const chunkSize = 500;
+        let searchesCompleted = {};
+        for (let i = 0; i < entries_to_update.length; i += chunkSize) {
+            const chunk = entries_to_update.slice(i, i + chunkSize);
+
+            let user_role = chunk[0].user_role;
+            let searchQuery = '';
+            for(let i=0; i<chunk.length; i++){
+                let entry = chunk[i];
+                if (i==0){
+                  searchQuery = `| inputlookup cs_user_inventory | search uuid IN (`;
```

Comment:
Can we make use of custom command? So we can avoid inputlookup and outputlookup with device inventory and user inventory?

Reply:
To update the lookup, we have to provide the same values for all the existing fields and values are in nested json format due to matching users. 
but custom command fetches the data by simplifying the json structure. 
and if we do with custom command then it is hard to decide which value pickup from multivalue field to again add it via custom command.

Reply:
We can pass uuid to the custom command and have a specific handler to handle the update is_previledged_user field right?

Similarly how we are making updates in Device Inventory. I'm trying to keep things consitent.

Reply:
Yes we can do by adding the specific handler in user inventory. I'll add that.